core/src/test/java/io/aiven/kafka/tieredstorage/manifest/SegmentManifestProviderTest.java

```diff
@@ -141,4 +142,24 @@ void shouldPropagateIOException(@Mock final InputStream isMock) throws StorageBa
             .isInstanceOf(IOException.class)
             .hasMessage("test");
     }
+
+    @Test
+    void shouldNotPoisonCacheWithFailedFutures()
+        throws StorageBackendException {
+        when(storage.fetch(MANIFEST_KEY))
+            .thenThrow(new StorageBackendException("test"))
+            .thenReturn(new ByteArrayInputStream(MANIFEST.getBytes()));
+
+        assertThatThrownBy(() -> provider.get(MANIFEST_KEY))
+            .isInstanceOf(StorageBackendException.class)
+            .hasMessage("test");
+
+        final var chunkIndex = new FixedSizeChunkIndex(100, 1000, 110, 110);
+        final var expectedManifest = new SegmentManifestV1(chunkIndex, SEGMENT_INDEXES, false, null, null);
+
+        await().atMost(Duration.ofMillis(50))
+            .pollInterval(Duration.ofMillis(5))
+            .ignoreExceptions()
+            .until(() -> provider.get(MANIFEST_KEY).equals(expectedManifest));
```

Comment:
Same as in ChunkCache: seems that awaitility is not needed since the second call to the mock I assume will always return the expectedManifest.

Reply:
> Same as in ChunkCache: seems that awaitility is not needed since the second call to the mock I assume will always return the expectedManifest.

Answered above: this assumption is incorrect. `.get`s in quick succession result in the more than one time return of the failed future. This is a result of the future scheduling.